metricbeat/module/mysql/mysql.go

```diff
@@ -24,6 +24,26 @@ func (m *Moduler) Setup(mo *helper.Module) error {
 	return nil
 }
 
+// CreateDSN creates a dsn string out of hostname, username and password
+func CreateDSN(host string, username string, password string) string {
+	// Example: [username[:password]@][protocol[(address)]]/
+
+	dsn := host
+
+	if username != "" || password != "" {
+		dsn = "@" + dsn
+	}
+
+	if password != "" {
+		dsn = ":" + password + dsn
```

Comment:
Just checked and the mysql library states that it does not need escaping: https://github.com/go-sql-driver/mysql#password But I didn't test it it works.